content/test-design.md

```diff
@@ -85,114 +85,93 @@ Also discuss why some are easier to test than others.
    ```
    ```{code-tab} c++
 
-
-   # 1
-   /*   Computes the factorial of n.   */
-   int factorial(int n)
-   {
-      if(n<0)
-        throw "received negative input";
-      if(n==1)
-        return 1;
-      else
-        return factorial(n-1) * n;
+   // 1
+   /* Computes the factorial of n recursively. */
+   constexpr unsigned int factorial(unsigned int n) {
+      return (n <= 1) ? 1 : (n * factorial(n - 1));
```

Comment:
nice to have it ternary and recursive, but the python and julia versions deliberately don't do it for simplicity reasons. I guess it doesn't matter much for this example though

Reply:
I also find ternaries a bit hard to read and I am avoiding these in my own code but I think it's OK here since it is a relatively simple example.

Reply:
While a non-recursive function would certainly work and would be OK C++, this example shows a recursion that can be evaluated at compile time because of the `constexpr` qualifier. The ternary `if` is necessary in C++11 `constexpr` function can only have one return statement (no loops, not multi-line conditionals)